casper/src/main/scala/coop/rchain/casper/api/BlockReportAPI.scala

```diff
@@ -16,7 +15,7 @@ import coop.rchain.shared.syntax._
 
 import scala.collection.concurrent.TrieMap
 
-class BlockReportAPI[F[_]: Concurrent: Metrics: EngineCell: Log: BlockStore](
+class BlockReportAPI[F[_]: Concurrent: BlockStore: Metrics: Log](
```

Comment:
Is there any rule to sort implicits here?

Reply:
I'm using the rule to first specify execution dependencies like Monad, Sync, Concurrent, Time, ... and at the end less important dependencies like Metrics, Log, ... And all else in the middle.

And with more arguments I found it's useful to disable formatting and add additional comments.
https://github.com/rchain/rchain/blob/65d20ccf0b97bf3f24df18d50ee9c30c8ce0f6e5/casper/src/main/scala/coop/rchain/casper/engine/Running.scala#L249-L255